### PR TITLE
test:fix:e2e Fix flaky PPOM Blockaid tests by adding more mock JSON-API requests

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

Discovered PPOM requests may fail resulting in "Request may not be safe" modal which breaks our checks when we expect PPOM to respond to a benign or malicious request.

Fixes Blockaid e2e flaky tests by mocking additional requests to avoid waiting for responses

Additional mocks added:
eth_feeHistory, eth_getTransactionCount, debug_traceCall, and eth_getBlockByNumber

comments:

1. it's unknown to me why the benign tests are heaps slower than the malicious tests run in this file. 
2. at some point I tested separating each test to get better logging. I reverted this to save test time.
  a. with cases separated
    ```
    <testsuite name="Mocha Tests" tests="10" failures="0" errors="0" skipped="0" timestamp="Wed, 18 Oct 2023 10:23:32 GMT" time="157.403">
      <testcase classname="Confirmation Security Alert - Blockaid" name="should show &#x22;Request may not be safe&#x22; if the PPOM request fails to check transaction" time="13.647"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for benign requests" name="should work for benign eth_sendTransaction with no value" time="18.148"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for benign requests" name="should work for benign eth_sendTransaction with value" time="18.019"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for benign requests" name="should work for benign Blur eth_sendTransaction" time="18.275"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for benign requests" name="should work for benign Seaport eth_sendTransaction" time="18.066"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for benign requests" name="should work for benign eth_signTypedData" time="18.139"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for malicious requests" name="should work for permit_farming reason type" time="13.076"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for malicious requests" name="should work for raw_native_token_transfer reason type" time="13.243"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for malicious requests" name="should work for seaport_farming reason type" time="13.394"/>
      <testcase classname="Confirmation Security Alert - Blockaid should show security alerts for malicious requests" name="should work for trade_order_farming reason type" time="13.388"/>
    </testsuite>
    ```
   b. without cases separated
    ```
    <testsuite name="Mocha Tests" tests="3" failures="0" errors="0" skipped="0" timestamp="Wed, 18 Oct 2023 10:15:35 GMT" time="95.358">
      <testcase classname="Confirmation Security Alert - Blockaid" name="should show security alerts for benign requests" time="65.598"/>
      <testcase classname="Confirmation Security Alert - Blockaid" name="should show security alerts for malicious requests" time="16.594"/>
      <testcase classname="Confirmation Security Alert - Blockaid" name="should show &#x22;Request may not be safe&#x22; if the PPOM request fails to check transaction" time="13.162"/>
    </testsuite>
    ```

## **Manual testing steps**

1. yarn build:test:flask
2. run tests e.g.
- yarn test:e2e:single test/e2e/flask/ppom-blockaid-alert.spec.js --browser=chrome --leave-running=true
- yarn test:e2e:single test/e2e/flask/ppom-malicious-erc20-approval.spec.js --browser=firefox --leave-running=true
- yarn test:e2e:single test/e2e/flask/ppom-malicious-erc20-transfer.spec.js --browser=firefox --leave-running=true

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1482

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by digiwand** [comment]
adding mocks for these two requests

    ['eth_feeHistory'],
    ['eth_getTransactionCount'],

 still errored out

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/59758/workflows/4a3d4a6e-cd55-487a-99f2-6b224169e302/jobs/1850234

**Comment by digiwand** [comment]
hm.. with the retry logic we still run into

```
[driver] Called 'switchToWindowWithTitle' with arguments ["MetaMask Notification"]
[driver] Called 'findElement' with arguments [{"css":"[data-testid=\"security-provider-banner-alert\"]","text":"Request may not be safe"}]
<testsuite name="Mocha Tests" tests="3" failures="0" errors="1" skipped="0" timestamp="Mon, 16 Oct 2023 19:57:06 GMT" time="150.487">
<testcase classname="Confirmation Security Alert - Blockaid" name="should not show security alerts for benign requests" time="68.16"/>
<testcase classname="Confirmation Security Alert - Blockaid" name="should show security alerts for malicious requests" time="69.114"><failure>Retry limit reached
Error: Retry limit reached
    at retry (development/lib/retry.js:45:6)
    at process.btnSelector (node:internal/process/task_queues:95:5)
    at /home/circleci/project/test/e2e/flask/ppom-blockaid-alert.spec.js:170:13
    at withFixtures (test/e2e/helpers.js:107:9)
    at Context.&#x3C;anonymous&#x3E; (test/e2e/flask/ppom-blockaid-alert.spec.js:91:9)</failure></testcase>
```

**Comment by digiwand** [comment]
even with retry logic, this test fails
```
[driver] Called 'switchToWindowWithTitle' with arguments ["MetaMask Notification"]
[driver] Called 'findElement' with arguments [{"css":"[data-testid=\"security-provider-banner-alert\"]","text":"Request may not be safe"}]
<testsuite name="Mocha Tests" tests="3" failures="0" errors="1" skipped="0" timestamp="Mon, 16 Oct 2023 19:57:06 GMT" time="150.487">
<testcase classname="Confirmation Security Alert - Blockaid" name="should not show security alerts for benign requests" time="68.16"/>
<testcase classname="Confirmation Security Alert - Blockaid" name="should show security alerts for malicious requests" time="69.114"><failure>Retry limit reached
Error: Retry limit reached
    at retry (development/lib/retry.js:45:6)
    at process.btnSelector (node:internal/process/task_queues:95:5)
    at /home/circleci/project/test/e2e/flask/ppom-blockaid-alert.spec.js:170:13
    at withFixtures (test/e2e/helpers.js:107:9)
    at Context.&#x3C;anonymous&#x3E; (test/e2e/flask/ppom-blockaid-alert.spec.js:91:9)</failure></testcase>
```

It was expecting "This is a deceptive request", but the test found "Request may not be safe". This implies that the PPOM request failed 🤔 

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/59877/workflows/8fac1195-dede-4452-a017-654e945fa3a5/jobs/1855809

test-failure-ss

**Comment by digiwand** [comment]
Thanks for looking into these missing mocks and suggesting adding debug_traceCall and eth_getBlockByNumber @seaona :) 
CI/CD tests have passed 4 times in a row. I think this is ready for review now

File: test/e2e/flask/ppom-blockaid-alert.spec.js [hunk 1]
```diff
@@ -31,7 +145,7 @@ async function mockInfura(mockServer) {
  * @see {@link https://wobbly-nutmeg-8a5.notion.site/MM-E2E-Testing-1e51b617f79240a49cd3271565c6e12d}
  */
 describe('Confirmation Security Alert - Blockaid', function () {
-  it('should not show security alerts for benign requests', async function () {
+  it('should show security alerts for benign requests', async function () {
```
Comment (digiwand) [inline test/e2e/flask/ppom-blockaid-alert.spec.js:148]:
```suggestion
  it('should not show security alerts for benign requests', async function () {
```

**Review by seaona — APPROVED** [review]
LGTM @digiwand :fire:

**Review by danjm — APPROVED** [review]

**Pull request merged**